src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -15,35 +15,34 @@
 public class HttpClientSettingsKey implements Serializable {
 
   private OCSPMode ocspMode;
-  private boolean useProxy;
+  private boolean useProxy = false;
   private String proxyHost = "";
   private int proxyPort = 0;
   private String nonProxyHosts = "";
   private String proxyUser = "";
   private String proxyPassword = "";
   private String proxyProtocol = "http";
+  private String userAgentSuffix = "";
 
   public HttpClientSettingsKey(
       OCSPMode mode,
+      boolean useProxy,
       String host,
       int port,
       String nonProxyHosts,
       String user,
       String password,
-      String scheme) {
-    this.useProxy = true;
+      String scheme,
+      String userPrefix) {
```

Comment:
Can just let this input param be named the same as the member. `userAgentSuffix`. Also, since it's a suffix, not a prefix, probably don't want to call it prefix here.